internal/go/cache/prog.go

```diff
@@ -384,15 +325,15 @@ func (c *ProgCache) Put(a ActionID, file io.ReadSeeker) (_ OutputID, size int64,
 		return OutputID{}, 0, err
 	}
 
-	if !c.can[cmdPut] {
+	if !c.can[cacheprog.CmdPut] {
 		// Child is a read-only cache. Do nothing.
 		return out, size, nil
 	}
 
-	res, err := c.send(c.ctx, &ProgRequest{
-		Command:  cmdPut,
+	res, err := c.send(c.ctx, &cacheprog.Request{
+		Command:  cacheprog.CmdPut,
 		ActionID: a[:],
-		ObjectID: out[:],
+		OutputID: out[:],
```

Comment:
In Go 1.24, they write both fields
```
OutputID: out[:],
ObjectID: deprecatedValue, // TODO(bradfitz): remove in Go 1.25
```
For backward compatibility I guess ?

Reply:
I removed the ObjectID, because this PR targets golangci-lint v2, we can be breaking.
Also, the programmatic cache is experimental, so breaking changes are expected.

Reply:
ok :+1: